addons/misra.py

```diff
@@ -1353,6 +1364,42 @@ def misra_7_3(self, rawTokens):
             if compiled.match(tok.str):
                 self.reportError(tok, 7, 3)
 
+    def misra_7_4(self, data):
+        # A string literal shall not be assigned to an object unless the object's type
+        # is constant.
+        def reportErrorIfVariableIsNotConst(variable, stringLiteral):
+            if variable.valueType:
+                if variable.valueType.constness == 0:
```

Comment:
I guess you want to warn for:

    char * const str = "x";

the constness is 2 for that.

Reply:
I want to warn when the const is not used in this case. So maybe constness != 2 is correct then.